community/modules/compute/schedmd-slurm-gcp-v6-nodeset/outputs.tf

```diff
@@ -32,6 +32,16 @@ output "nodeset" {
     EOD
   }
 
+  precondition {
+    condition     = var.accelerator_topology == null || (var.accelerator_topology != null) == strcontains(split("-", var.machine_type)[2], "g")
+    error_message = "accelerator_topology can only be used by a machine with gpus"
+  }
+
+  precondition {
+    condition     = (var.accelerator_topology == null) || (try(tonumber(split("x", var.accelerator_topology)[1]), 0) % tonumber(replace(split("-", var.machine_type)[2], "g", "")) == 0)
```

Comment:
We have a better way to get number of GPUs, please take a look at `guest_accelerators`